openshift/deploy.sh

```diff
@@ -1,17 +1,116 @@
 #!/usr/bin/bash -e
 
-# Note this script assumes that config maps and secrets are already deployed
+#Load configuration from env variables
+source env.sh
+
+#Check for required tools
+if ! [ -x "$(command -v aws)" ]; then
+  echo 'Error: aws is not installed.' >&2
+  exit 1
+fi
+
+if ! [ -x "$(command -v psql)" ]; then
+  echo 'Error: psql is not installed.' >&2
+  exit 1
+fi
+
+if ! [ -x "$(command -v oc)" ]; then
+  echo 'Error: oc is not installed.' >&2
+  exit 1
+fi
+
+function generate_and_deploy_config() {
+  oc process -p DEPLOYMENT_PREFIX="${OC_PROJECT}" \
+             -p KEYCLOAK_URL="${KEYCLOAK_URL}" \
+             -f ${here}/config-template.yaml > ${here}/config.yaml
+  oc apply -f config.yaml
+}
+
+function deploy_secrets() {
+  #All secrets must be base64 encoded
+  oc process -f secrets-template.yaml -p AWS_ACCESS_KEY_ID=`echo -n $AWS_ACCESS_KEY_ID | base64` \
+                                      -p AWS_SECRET_ACCESS_KEY=`echo -n $AWS_SECRET_ACCESS_KEY | base64` \
+                                      -p GITHUB_API_TOKENS=`echo -n $GITHUB_API_TOKENS | base64` \
+                                      -p GITHUB_OAUTH_CONSUMER_KEY=`echo -n $GITHUB_OAUTH_CONSUMER_KEY | base64` \
+                                      -p GITHUB_OAUTH_CONSUMER_SECRET=`echo -n $GITHUB_OAUTH_CONSUMER_SECRET | base64` \
+                                      -p FLASK_APP_SECRET_KEY=`echo -n $FLASK_APP_SECRET_KEY | base64` \
+                                      -p RDS_ENDPOINT="`echo -n $RDS_ENDPOINT | base64`" \
```

Comment:
`RDS_ENDPOINT` can be a long string, it would be safer to use `echo -n $RDS_ENDPOINT | base64 -w 0` here.